### PR TITLE
Add matcher for resources

### PR DESCRIPTION
**Pull request by Blackbaud-JasonBodnar**
#16

### CODE REVIEW

File: src/app/public/matchers/matchers.ts [hunk 1]
```diff
@@ -163,6 +170,38 @@ const matchers: jasmine.CustomMatcherFactories = {
...
+  toHaveResourceText(): jasmine.CustomMatcher {
+    return {
+      compare(
+        actual: string,
+        name: string,
+        args: { name: string, args: any[] },
+        callback: () => void = () => {}
+      ): jasmine.CustomMatcherResult {
+
+        let skyAppResourcesService: SkyAppResourcesService = TestBed.get(SkyAppResourcesService);
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/matchers/matchers.ts]: Curious, but what happens if the consumer isn't using `SkyAppResourcesService` and they run this matcher?

File: src/app/public/matchers/matchers.spec.ts [hunk 2]
```diff
@@ -8,10 +8,26 @@ import {
...
-import { expect } from './matchers';
+import {
+  SkyAppResourcesService,
+  SkyI18nModule
+} from '@skyux/i18n';
+
+import {
+  EMPTY,
+  of
+} from 'rxjs';
```
[the remark this answers is not in the document]
Comment (Blackbaud-JasonBodnar) [reply src/app/public/matchers/matchers.spec.ts]: When I make that change I get:

```
ERROR in [at-loader] ./src/app/public/matchers/matchers.spec.ts:16:3 
    TS2305: Module '"../../../../node_modules/rxjs/add/observable/of"' has no exported member 'EMPTY'.

ERROR in [at-loader] ./src/app/public/matchers/matchers.spec.ts:17:3 
    TS2305: Module '"../../../../node_modules/rxjs/add/observable/of"' has no exported member 'of'.
```

File: src/app/public/matchers/matchers.ts (hunk 1 above)
Comment (Blackbaud-JasonBodnar) [reply src/app/public/matchers/matchers.ts]:
```
Error: StaticInjectorError[SkyAppResourcesService]: 
	  NullInjectorError: No provider for SkyAppResourcesService!
```

which is exactly what you'd get if you tried to use it directly in a test without listing it as a provider.

**Comment by Blackbaud-JasonBodnar** [comment]
I just pushed some changes that renamed the matcher I created to `toEqualResourceText()` since it takes an string and added a new matcher called `toHaveResourceText()` that takes an element and is analogous to the existing `toHaveText()` matcher.

**Comment by Blackbaud-SteveBrush** [comment]
@Blackbaud-JasonBodnar Thanks for the PR! I'm going to move this PR to an official branch so that Travis-CI will run all the appropriate steps.

**Comment by Blackbaud-SteveBrush** [comment]
Closing in favor of https://github.com/blackbaud/skyux-sdk-testing/pull/18

**Pull request closed without merging**